### PR TITLE
Set KEEPALIVE and NO_DELAY socket options on peer connections

### PR DESCRIPTION
**Pull request by alexlarsson**

### CODE REVIEW

File: src/agent/agent.c [hunk 1]
```diff
@@ -1029,6 +1029,16 @@ bool agent_start(Agent *agent) {
                 return false;
         }
 
+        r = bus_socket_set_no_delay(agent->peer_dbus);
+        if (r < 0) {
+                hirte_log_warn("Failed to set NO_DELAY on socket");
+        }
+
+        r = bus_socket_set_no_delay(agent->peer_dbus);
```
Comment (rhatdan) [inline src/agent/agent.c]: This looks like you have a cut and paste error? You set no delay twice?

File: src/manager/manager.c [hunk 2]
```diff
@@ -367,6 +368,15 @@ static int manager_accept_node_connection(
                 return -1;
         }
 
+        int r = bus_socket_set_no_delay(dbus_server);
+        if (r < 0) {
+                hirte_log_warn("Failed to set NO_DELAY on socket");
+        }
+        r = bus_socket_set_no_delay(dbus_server);
```
Comment (rhatdan) [inline src/manager/manager.c]: Cut and paste error again.

**Review by rhatdan — APPROVED** [review]

File: src/agent/agent.c (hunk 1 above)
Comment (pypingou) [reply src/agent/agent.c]:
```suggestion
        r = bus_socket_set_keepalive(agent->peer_dbus);
```

File: src/manager/manager.c (hunk 2 above)
Comment (pypingou) [reply src/manager/manager.c]:
```suggestion
        r = bus_socket_set_keepalive(dbus_server);
```

**Comment by pypingou** [comment]
We've shaved ~0.1 seconds with this change (adjusted for keepalive)

File: src/libhirte/bus/utils.c [hunk 3]
```diff
@@ -164,3 +167,33 @@ char *bus_path_escape(const char *s) {
...
+}
+
+int bus_socket_set_keepalive(sd_bus *bus) {
+        int fd = sd_bus_get_fd(bus);
+        if (fd < 0) {
+                return fd;
+        }
+
+        int flag = 1;
+        int r = setsockopt(fd, SOL_SOCKET, SO_KEEPALIVE, (char *) &flag, sizeof(int));
```
Comment (sdunnagan) [inline src/libhirte/bus/utils.c:275]: TCP keepalives have a default of 2 hours.  On my Fedora host:
```
$ cat /proc/sys/net/ipv4/tcp_keepalive_time
7200
```
Keepalive time can be set for all TCP sockets on the system:
```
$ echo 10 > /proc/sys/net/ipv4/tcp_keepalive_time
```

But that is for all sockets, system wide, which may not be what we want.

I read that there is a ```TCP_KEEPIDLE``` option for setting the keepalive interval
only for the current socket.  So after setting the socket with ```SO_KEEPALIVE```,
we could next use ```TCP_KEEPIDLE``` to set the interval to for example 10 seconds:
```
  int keepidle = 10;
  setsockopt(sock, IPPROTO_TCP, TCP_KEEPIDLE, &keepidle, sizeof(int));
```

**Review by sdunnagan — APPROVED** [review]

File: src/libhirte/bus/utils.c (hunk 3 above)
Comment (alexlarsson) [reply src/libhirte/bus/utils.c:275]: I set KEEPIDLE and KEEPINTVL both to 10.

**Pull request merged**